lib/git/lib.rb

```diff
@@ -571,6 +574,7 @@ def commit(message, opts = {})
       arr_opts << "--author=#{opts[:author]}" if opts[:author]
       arr_opts << "--date=#{opts[:date]}" if opts[:date].is_a? String
       arr_opts << '--no-verify' if opts[:no_verify]
+      arr_opts |= opts[:raw_opts] if opts.key?(:raw_opts)
```

Comment:
I think about it and decide that `:allow_empty_message` will be really rare used, at the other hand `:raw_opts` can be used for different purposes

>Also, isn't the documentation for opts wrong? Should it say that a Hash is expected instead of an Array?

Year, it definitely Hash, should we fix it in this Pull Request?

Reply:
I believe we should change this to follow the convention of using specific hash keys for passing options (passing `:allow_empty_message` instead of `:raw_opts`)

One reason for this convention is to constrain how use this gem is used so users don't give combinations of options that this gem wouldn't respond well to. This is how the rest of the Git gem interface works. I don't see a reason to violate this convention just for this case. After all, isn't `:allow_empty` just as rarely used?  Why give two different ways for passing arguments?

Reply:
Ok, you convince me